tests/kernel/device/CMakeLists.txt

```diff
@@ -14,3 +14,9 @@ target_sources(app PRIVATE ${app_sources})
 # at all. So add this flag to merge string constants such that
 # the pointer comparison would be exercised.
 zephyr_cc_option_ifdef(CONFIG_COVERAGE -fmerge-constants)
+
+zephyr_syscall_header(src/abstract_driver.h)
+zephyr_iterable_section(
+  NAME abstract_driver_api
+  KVMA RAM_REGION GROUP RODATA_REGION
+  SUBALIGN CONFIG_LINKER_ITERABLE_SUBALIGN)
```

Comment:
this looks like too much to define a new driver class

Reply:
Yes, but it is needed to support the `CMAKE_LINKER_GENERATOR`, and as [commented](https://github.com/zephyrproject-rtos/zephyr/pull/71773#pullrequestreview-2087594009) by @tejlmand we should eventually generate it from scripting in a follow-up PR.

Reply:
I really don't want to introduce a new feature that forces users to write this for every single driver class, less before an LTS release. This should be fixed before. Ref. https://github.com/zephyrproject-rtos/example-application/pull/65

Reply:
Hey @gmarull,

Trying to get this moving again, now that 3.7 is released.
It has been discussed in the arch meeting group, it only has to be done once for a driver class, which was fine to start with.

If you feel this is a blocker, but don't have another proposal, this PR will be stuck.

Reply:
Yes, to me, this is a blocker. As a driver class developer, I should not deal with such level of impl detail, ie

```
zephyr_iterable_section(
  NAME blink_driver_api
  KVMA RAM_REGION GROUP RODATA_REGION
  SUBALIGN CONFIG_LINKER_ITERABLE_SUBALIGN)
```

Reply:
I agree.

Reply:
It should only be done once for a class, and can be checked during review. Also CI can catch it if we add more test cases with `CMAKE_LINKER_GENERATOR`.

Nevertheless, I'd prefer to have it generated too, but have no clear solution for it.

Adding a "helper" macro with less arguments makes it even more confusing IMO.

Reply:
Anything that involves ELF hacking is a -1 from my side as well. The need for ELF hacking to obtain a valid build is a design failure to me.